compiler/rustc_resolve/src/build_reduced_graph.rs

```diff
@@ -913,14 +914,21 @@ impl<'a, 'b, 'tcx> BuildReducedGraphVisitor<'a, 'b, 'tcx> {
             let entry = self
                 .r
                 .extern_prelude
-                .entry(ident.normalize_to_macros_2_0())
+                .entry(ident)
                 .or_insert(ExternPreludeEntry { binding: None, introduced_by_item: true });
-            // Binding from `extern crate` item in source code can replace
-            // a binding from `--extern` on command line here.
-            entry.binding = Some(imported_binding);
             if orig_name.is_some() {
                 entry.introduced_by_item = true;
             }
+            // Binding from `extern crate` item in source code can replace
+            // a binding from `--extern` on command line here.
+            if ident.name != kw::Underscore && entry.binding.is_some() {
```

Comment:
`ident.name != kw::Underscore` is to ensure the following code works well.

```rs
extern core as _;
extern core as _;
fn main() {}

Reply:
It's clear that there should be no `span_delayed_bug` in this case, but we probably shouldn't do `entry.binding = Some(imported_binding)` either because `_` can still (?) result in the "inconsistent resolution" ICE.

Reply:
Nice catch, I had forgotten that use ::_ is also valid code.